docs/proposals/cluster-backup-integration.md

```diff
@@ -0,0 +1,104 @@
+# Cluster Backup Integration
+**Author**: @ahmadhamzh @moelsayed
+
+**Status**: Draft proposal
+
+## Goals
+Provide an automated backup solution for KKP user clusters, based on [velero](https://velero.io/). The velero integration solution should initially support the following use cases:
+- Specific user cluster applications and/or namespaces backup and recovery, including volumes.
+- User cluster disaster recovery (e.g. restore accidentally deleted clusters).
+- User cluster migration to different KKP deployment, DC or seed.
+
+In addition to this, the following basic features should be supported:
+- Define backup destinations per provider.
+- Select backup destination for backup configuration.
+- Define scheduled and one time backups.
+- Delete backups/backup configuration
+- Define automatic backup expiration/rotation.
+- Provide sufficient UI interface elements to support these requirements.
+
+The initial implementation will target support for the Openstack Provider.
+
+## Non-Goals
+- Implement a cluster backup solution from scratch.
+- Surface all velero features and capabilities through the UI.
+- Support for community maintained cloud providers is out of scope.
+- Managing and provisioning storage providers configuration.
+- Backup and restore hooks.
+- Backup and restore of Seed cluster objects are out of scope.
+- Validation of restored resources is out of scope. 
+
+ ## Motivation and Background
+Our current backup implementation is focused on user cluster etcd backup only. It's designed as a recovery mechanism to improve the reliability of the user cluster control plane. This doesn't allow user to fine tune their backup scope to specific applications, namespaces or labels. It doesn't backup user cluster application data (e.g. volumes)
+
+Additionally, the etcd backups are tightly coupled to the user cluster life-cycle. It doesn't support full cluster restores and/or cluster migration.
+
+With velero integration, user will be able to create more granular backup configuration with focus on application level data and with persistent volume backup support. It will also allow for disaster recovery and migration scenarios that are not currently supported by the existing solution.
+
+## Implementation
+
+Initially, the integration will be modeled on the UI and workflow of the current etcdbackup feature.
+
+The integration will be enabled by default for newly provisioned clusters only. For existing user cluster, it must be manually enabled. 
+
+No default backup configuration will be created for newly created clusters. Admins or project/cluster owners will be required to create backup configuration and point them to a seed level backup destination.
```

Comment:
> Admins or project/cluster owners will be required to create backup configuration

Project owners should be able to create backup configurations.

The question is, who should be able to set up backup destinations to the providers (admins or project owners)? 

> and point them to a seed level backup destination.

Why is the backup destination seed level?

Reply:
>The question is, who should be able to set up backup destinations to the providers (admins or project owners)?
I am not sure. What is the current workflow for etcdbackup?

>Why is the backup destination seed level?
This is not a design decision. My understanding is that currently seed/DC level in KKP?

Reply:
So backups will be a Project level feature, but presets for the storage spaces of the backups can be both on Project and the Admin level, it has not been fully decided yet. For example etc backups has this on the Admin level. Please check the possibility of both doing it (creating and deleting presets for storage spaces) from Project level and Admin level. If both of them is truly feasible, then we make a decision.

To the other question: did you mean that the presets for the storage spaces of the backups would be stored on the seed level? Did I get it correctly? Basically I just don't really understand what the original sentence means.